media/js/firefox/australis/fx38_0_5/firstrun.js

```diff
@@ -0,0 +1,149 @@
+            // GA track video play
+            window.dataLayer.push({
+                'event': 'firstrun-38.0.5-video',
+                'interaction': 'start',
+                'videoTitle': 'When its Personal Campaign Video'
+            });
+        }).on('ended', function() {
+            // GA track video finish
+            window.dataLayer.push({
+                'event': 'firstrun-38.0.5-video',
+                'interaction': 'finish',
+                'videoTitle': 'When its Personal Campaign Video'
+            });
+
+            // take a little breath before closing modal
+            setTimeout(function() {
+                Mozilla.Modal.closeModal();
+            }, 500);
+        });
+    }
+
+    // display appropriate copy
+    if (!ctaDirect) {
+        $('.ctaMenu').addClass('visible');
+    } else {
+        $('.ctaDirect').addClass('visible');
+    }
+
+    var showVideo = function(origin, autoplay) {
+        var opts = {
+            title: $videoTitle.text()
+        };
+
+        if (autoplay) {
+            opts.onCreate = function() {
+                // slight pause after modal opens
+                setTimeout(function() {
+                    $video[0].play();
+                }, 250);
+            };
+        }
+
+        Mozilla.Modal.createModal(origin, $videoFrame, opts);
+    };
+
+    // to be safe, make sure user is on desktop firefox version 38 or higher
+    if (window.isFirefox() && !window.isFirefoxMobile() && window.getFirefoxMasterVersion() >= 38) {
+        // if primary CTA should highlight Sync in settings
+        if (!ctaDirect) {
+            // Query if the UITour API is working before we start the tour
+            Mozilla.UITour.getConfiguration('sync', function() {
+                $document = $(document);
+
+                $ctaSignup.on('click', function(e) {
+                    e.preventDefault();
+
+                    // call twice two correctly position highlight
+                    // https://bugzilla.mozilla.org/show_bug.cgi?id=1049130
+                    Mozilla.UITour.showHighlight('accountStatus', 'wobble');
+                    Mozilla.UITour.showHighlight('accountStatus', 'wobble');
+
+                    // allow clicking anywhere in page to hide menu
+                    // behind a timeout so event isn't captured with *this* click
+                    setTimeout(function() {
+                        $document.one('click.hideHighlight', function() {
+                            Mozilla.UITour.hideHighlight();
+                        });
+                    }, 50);
+                });
+
+                $document.on('visibilitychange', function() {
+                    if (document.hidden) {
+                        $document.off('click.hideHighlight');
+                        Mozilla.UITour.hideHighlight();
+                    }
+                });
+            });
+        } else {
+            $ctaSignup.on('click', function(e) {
```

Comment:
Should this even handler be behind a call to `Mozilla.UITour.ping()`, so it only gets added if UITour is working?